src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.RequestTelemetryLogger.cs

```diff
@@ -52,28 +62,57 @@ public void LogFailure(string methodName)
             /// </summary>
             public void Dispose()
             {
+                if (_histogramLogAggregator is null || _histogramLogAggregator.IsEmpty)
+                {
+                    return;
+                }
+
+                var queuedDurationCounter = _histogramLogAggregator.GetValue(QueuedDurationKey);
+                Logger.Log(FunctionId.LSP_TimeInQueue, KeyValueLogMessage.Create(LogType.Trace, m =>
+                {
+                    m["server"] = _serverTypeName;
+                    m["bucketsize"] = queuedDurationCounter?.BucketSize;
+                    m["maxbucketvalue"] = queuedDurationCounter?.MaxBucketValue;
+                    m["buckets"] = queuedDurationCounter?.GetBucketsAsString();
+                }));
+
                 foreach (var kvp in _requestCounters)
                 {
+                    var requestExecutionDuration = _histogramLogAggregator.GetValue(kvp.Key);
+
                     Logger.Log(FunctionId.LSP_RequestCounter, KeyValueLogMessage.Create(LogType.Trace, m =>
                     {
                         m["server"] = _serverTypeName;
                         m["method"] = kvp.Key;
-                        m["successful"] = kvp.Value._successfulCount;
+                        m["successful"] = kvp.Value._succeededCount;
                         m["failed"] = kvp.Value._failedCount;
                         m["cancelled"] = kvp.Value._cancelledCount;
+                        m["bucketsize"] = requestExecutionDuration?.BucketSize;
+                        m["maxbucketvalue"] = requestExecutionDuration?.MaxBucketValue;
+                        m["buckets"] = requestExecutionDuration?.GetBucketsAsString();
                     }));
                 }
 
                 // Clear telemetry we've published in case dispose is called multiple times.
                 _requestCounters.Clear();
+                _histogramLogAggregator = null;
             }
 
+            private static HistogramLogAggregator CreateLogAggregator() => new HistogramLogAggregator(bucketSize: 50, maxBucketValue: 5000);
```

Comment:
what does this mean?

Reply:
This is creating a histogram based aggregation - we put requests into buckets with size 50ms, with buckets from 0-5000
(so 0-50, 50-100, 100-150 etc)